packages/smooth_app/lib/cards/expandables/attribute_list_expandable.dart

```diff
@@ -32,20 +32,23 @@ class AttributeListExpandable extends StatelessWidget {
     for (final String attributeTag in attributeTags) {
       Attribute attribute =
           UserPreferencesModel.getAttribute(product, attributeTag);
-      chips.add(AttributeChip(attribute, width: iconWidth));
-      if (attribute != null &&
-          attribute.id == UserPreferencesModel.ATTRIBUTE_ADDITIVES) {
-        // TODO(monsieurtanuki): remove that cheat when additives are more standard
-        final List<String> additiveNames = product.additives?.names;
-        attribute = Attribute(
-          id: attribute.id,
-          title: attribute.title,
-          iconUrl: attribute.iconUrl,
-          descriptionShort:
-              additiveNames == null ? '' : additiveNames.join(', '),
-        );
+
+      // Some attributes selected in the user preferences might be unavailable for some products
+      if (attribute != null) {
+        chips.add(AttributeChip(attribute, width: iconWidth));
+        if (attribute.id == UserPreferencesModel.ATTRIBUTE_ADDITIVES) {
```

Comment:
Just a suggestion: I found the same bug this morning, and thought it was appropriate to display the attribute even if it's `null`:
```dart
if (attribute == null) {
  attribute = userPreferencesModel.getReferenceAttribute(attributeTag);
  attribute = Attribute(
    id: attribute.id,
    title: attribute.name,
    iconUrl: '',
    descriptionShort: 'no data',
  );
}
```

At least we can see: hey, I want to see the eco-score, but it's not available.
Not: hey, why can I see the eco-score only for some products?

Reply:
OK, we can do that. I don't see the getReferenceAttribute method though.
Do you have running code already for that?

Reply:
The whole fix is embedded in #134. I guess we can merge this one (#132), then when #134 is merged it will partly override #132 (which anyway has other interesting fixes that I reviewed).